bot/decorators.py

```diff
@@ -45,6 +45,45 @@ def predicate(ctx: Context) -> bool:
     return commands.check(predicate)
 
 
+def not_in_blacklist(
+    *,
+    channels: t.Container[int] = (),
+    categories: t.Container[int] = (),
+    roles: t.Container[int] = (),
+    override_roles: t.Container[int] = (),
+    redirect: t.Optional[int] = Channels.bot_commands,
+    fail_silently: bool = False,
+) -> t.Callable:
+    """
+    Check if a command was not issued in a blacklisted context.
+
+    The blacklists that can be provided are:
+
+    - `channels`: a container with channel ids for blacklisted channels
+    - `categories`: a container with category ids for blacklisted categories
+    - `roles`: a container with role ids for blacklisted roles
+
+    If the command was invoked in a context that was blacklisted, the member is either
+    redirected to the `redirect` channel that was passed (default: #bot-commands) or simply
+    told that they're not allowed to use this particular command (if `None` was passed).
+
+    The blacklist can be overridden through the roles specified in `override_roles`.
+    """
+    def predicate(ctx: Context) -> bool:
+        """Check if command was issued in a blacklisted context."""
+        not_blacklisted = not in_whitelist_check(ctx, channels, categories, roles, fail_silently=True)
+        overridden = in_whitelist_check(ctx, roles=override_roles, fail_silently=True)
+
+        success = not_blacklisted or overridden
+
+        if not success and not fail_silently:
+            raise InWhitelistCheckFailure(redirect)
```

Comment:
Yeah, this should be a new, distinct exception type.

Reply:
So just copy the code from the lib with a different class name? I guess that's what's been bugging me.

Reply:
You can make it a base class and make the whitelist/blacklist types subclasses with empty class bodies.

Reply:
Hmm the whitelist type is a part of the dpy library, I can't really edit that.

Reply:
It's not. It's right here https://github.com/python-discord/bot/blob/main/bot/utils/checks.py#L23

Reply:
Oh, woops